drivers/ethernet/Kconfig.adin2111

```diff
@@ -49,6 +49,24 @@ config ETH_ADIN2111_SPI_CFG0
 	  Must be set when ADIN uses 8-bit CRC (Generic SPI)
 	  or Protection Mode (OPEN Alliance) on the SPI Host Interface.
 
+DT_ETH_ADIN2111_PATH := $(dt_nodelabel_path,adin2111)
+DT_ETH_ADIN2111_SPI_OA := $(dt_node_bool_prop,$(DT_ETH_ADIN2111_PATH),spi-oa)
+
+config ETH_ADIN2111_SPI_OA
+	bool "SPI_OA"
+	default $(DT_ETH_ADIN2111_SPI_OA)
+	help
+	  Must be set when Open Alliance SPI is configured by HW bootstrap deep-switches.
```

Comment:
Why is this needed? Can't we just use `DT_INST_NODE_HAS_PROP` or equivalent in the code without going through one additional Kconfig symbol?

Reply:
This was an intermediate choice, since there is the CRC option (CFG0) actually in Kconfig. In this way all options are visible also in menuconfig.:

(2048) Stack size for a thread that processes ADIN IRQ
(2) Priority for internal incoming packet handler
(100) IP buffer timeout
[*] SPI_CFG0
[*] SPI_OA
[ ] SPI_OA_PROTECTION

In a next step i can set all the options in devicetree only, eventually.

Reply:
This is totally redundant and while there could be overlapping between DT and Kconfig it is alway better to just put the hardware configuration in DT when possible.

Intermediate steps are only going to increase the workload for all the parts involved (you, reviewers, maintaners, etc...) so I'd just suggest to move all the options into the DT already in this PR.

Reply:
@spectrum70 my mistake was to place the CRC option into Kconfig.

Ideally, the CRC should be moved into the DT. Some adin eval boards were recently added; and there it might be necessary to control CRC on/off at runtime (_due to board config switches_).

Reply:
Ok i proceed as required.